zookeeper-jute/src/main/java/org/apache/jute/BinaryInputArchive.java

```diff
@@ -85,6 +85,18 @@ public String readString(String tag) throws IOException {
     }
     
     static public final int maxBuffer = Integer.getInteger("jute.maxbuffer", 0xfffff);
+    static public final int readExtraSize;
+
+    static {
+        Integer configuredValue =
+            Integer.getInteger("zookeeper.jute.maxbuffer.extrasize", maxBuffer);
+        if (configuredValue < 1024) {
+            // Earlier hard coded value is 1024, So the value should not be less than that
+            readExtraSize = 1024;
```

Comment:
Do we have a reasonable estimate of an upper bound of the extra field padding which could reasonably cover more practice use cases, so users don't have to configure this value themselves? Obviously it's not tractable to find a single value that cover all cases (unless we use infinity), but it seems reasonable to increase this value - say by 10x while still be safeguarding requests with unreasonable length.

Reply:
In this patch we have changed the default value of extra size to same as max buffer value.  I do not foresee any scenario where record extra size is more than the actual record size.
	So this value is going to be sufficient and possibly there would not be any need to configure jute.maxbuffer.extrasize.